code/game/gamemodes/wizard/artefact.dm

```diff
@@ -306,4 +306,31 @@
 				return
 			T = get_turf(src)
 			if(user.z != T.z || is_holder_of(user, src))
-				user.dust()
\ No newline at end of file
+				user.dust()
+
+/obj/item/clothing/shoes/blindingspeed
+	name = "boots of blinding speed"
+	desc = "Blinds you while moving."
+	icon_state = "blindingspeed"
+	item_state = "blindingspeed"
+	wizard_garb = 1
+	var/speed_modifier = 4
+
+/obj/item/clothing/shoes/blindingspeed/equipped(mob/living/carbon/human/H, equipped_slot)
+	..()
+	if(istype(H) && H.get_item_by_slot(slot_shoes) == src && equipped_slot != null && equipped_slot == slot_shoes)
+		H.movement_speed_modifier *= speed_modifier
+		//H.change_sight(adding = BLIND)
+
+
+/obj/item/clothing/shoes/blindingspeed/step_action()
+	var/mob/living/carbon/human/H = loc
+	H.change_sight(adding = BLIND)
+	spawn(3)
```

Comment:
surely there's a better way that doesn't spawn every time you take a step

Reply:
There might be, but I'm not sure there's a check that happens when you stop moving.

Reply:
The invisibility genetic mutation starts doing its thing when you stand still, maybe you could look at how that does it.

Reply:
Invis relegates its stuff to Life(), which takes 0-2 seconds or so to happen. I did a little more testing, and 0-2 seconds doesn't feel unacceptably terrible compared to .5 seconds or so after you stop.